code/controllers/hooks.dm

```diff
@@ -1,39 +1,58 @@
+//* This file is explicitly licensed under the MIT license. *//
+//* Copyright (c) 2025 Citadel Station Developers           *//
+
 /**
- * @file hooks.dm
- * Implements hooks, a simple way to run code on pre-defined events.
+ * Typed hooks system.
```

Comment:
this is the key of why i'm replacing the old one, btw

i don't like untyped procs at all. or weird 'call all procs on this datum'. the byond compiler won't save you most of the time but the times it can is the reason why i want typed procs with parents. that, or, atleast the linter can do something about it & find all implementations work.